test/shmemx/query_initialized.c

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright (c) 2024 Intel Corporation. All rights reserved.
+ *  This software is available to you under the BSD license below:
+ *
+ *      Redistribution and use in source and binary forms, with or
+ *      without modification, are permitted provided that the following
+ *      conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdio.h>
+#include <shmem.h>
+
+int main(int argc, char* argv[])
+{
+    int initialized;
+    int init_count = 3;
+    int finalize_count = 3;
+    int inits_done = 0;
+    int finalizes_done = 0;
+
+    // Check if SHMEM is initialized before calling init
+    shmem_query_initialized(&initialized);
```

Comment:
I believe this should probably be `shmemx_query_initialized`...  At least for tests-sos v1.5.x.

If we go that route, then this test wouldn't build against your SOS implementation of course, but I think that's ok temporarily during our transition to v1.6.x.

I think let's keep the test this way (`shmem_`) until we are ready to merge https://github.com/Sandia-OpenSHMEM/SOS/pull/1144.  Then we can test against this?  Although, I'm not certain our CI covers shmemx very well... will have to check.

Sorry for the confusion...